pkg/server/handler/v1alpha1/integration.go

```diff
@@ -55,7 +57,35 @@ func ListIntegrations(ctx context.Context, tenant string, pagination *types.Pagi
 		integrations = append(integrations, *integration)
 	}
 
-	return types.NewListResponse(int(size), integrations[pagination.Start:end]), nil
+	results := []api.Integration{}
+	// Handle filter.
+	if qp.Filter == "" {
```

Comment:
For such case, return directly, would make the code clean.

if qp.Filter == "" {
  return ...
} 

...